lib/robot.js

```diff
@@ -144,19 +144,44 @@ class Robot {
       logger: log.child({installation: id})
     })
 
-    if (id) {
-      const res = await this.cache.wrap(`app:${id}:token`, () => {
-        log.trace(`creating token for installation`)
-        github.authenticate({type: 'integration', token: this.app()})
+    if (process.env.AUTH_METHOD === 'githubapp') {
+      if (id) {
+        const res = await this.cache.wrap(`app:${id}:token`, () => {
+          log.trace(`creating token for installation`)
+          github.authenticate({type: 'integration', token: this.app()})
 
-        return github.apps.createInstallationToken({installation_id: id})
-      }, {ttl: 60 * 59}) // Cache for 1 minute less than GitHub expiry
+          return github.apps.createInstallationToken({installation_id: id})
+        }, {ttl: 60 * 59}) // Cache for 1 minute less than GitHub expiry
 
-      github.authenticate({type: 'token', token: res.data.token})
+        github.authenticate({type: 'token', token: res.data.token})
+      } else {
+        github.authenticate({type: 'integration', token: this.app()})
+      }
     } else {
-      github.authenticate({type: 'integration', token: this.app()})
+      github = new GitHubApi({
```

Comment:
this likely needs to be migrated to `github.apps.createInstallationToken(...)` now. @bkeepers thoughts?

Reply:
nope, truly just not catching up on the last master-merge i did... 🤦‍♂️